File: nomad-samples/python/basic-live-channel-1.1/account/logout.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

from libraries import requests, json
'''
 * Logout
 *
'''
async def logout(USER_SESSION_ID, AUTH_TOKEN):
    HEADERS = {
        "Content-Type": "application/json",
        "Authorization": "Bearer " + AUTH_TOKEN
    }

    BODY = {
        "userSessionId": USER_SESSION_ID
    }

    try:
        # Send the request
        RESPONSE = requests.post(PUBLIC_URL + "/account/logout", headers= HEADERS, data = json.dumps(BODY))

        if RESPONSE.status_code != 200:
            raise Exception()
        
        return True

    except:
        await api_exception_handler(RESPONSE, "Logout failed")

File: nomad-samples/js/account-authenticaton/account/logout.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

/**
 * Logout
 *
 */
export default async function logout(APPLICATION_ID) {
    const HEADERS = new Headers();
    HEADERS.append("Content-Type", "application/json");
    HEADERS.append("Authorization", `Bearer ${sessionStorage.getItem("token")}`);

    // Build the POST payload body
    const BODY = {
       "userSessionId": sessionStorage.getItem("userSessionId"),
       "applicationId": APPLICATION_ID
    };

    // Send the request
    const response = await fetch(`${prjConstants.PORTAL_API_URL}/account/logout`, {
        method: "POST",
        headers: HEADERS,
        body: JSON.stringify(BODY)
    }).catch((exception) => {
        throw exception;
    });


    // Check for success
    if (response && response.ok) {
        // Get interval ID
        const intervalId = sessionStorage.getItem("intervalId");

        // Clear interval
        if (intervalId) {
            clearInterval(intervalId);
        }

        sessionStorage.clear();

        return true;
    }

    await apiExceptionHandler(response, "Logout failed");
}
